### PR TITLE
Rename BackwardMode.FULL --> UNROLL and simplify backward mode config

### PR DESCRIPTION
**Pull request by luisenp**
We can now pass strings to specify backward modes, instead of having to pass the specific enum type.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -141,7 +141,7 @@ outer_optimizer = torch.optim.RMSprop([phi], lr=0.001)
 for epoch in range(10):
     solution, info = layer.forward(
         input_tensors={"x": phi.clone(), "v": torch.ones(1, 1)},
-        optimizer_kwargs={"backward_mode": th.BackwardMode.IMPLICIT})
+        optimizer_kwargs={"backward_mode": "implicit"})
```
Comment (mhmukadam) [inline README.md:144]: We should update this example in the paper and the website following this.

File: theseus/optimizer/nonlinear/tests/test_backwards.py [hunk 2]
```diff
@@ -111,7 +111,7 @@ def fit_x(data_x_np):
         optimizer_kwargs={
             "track_best_solution": True,
             "verbose": False,
-            "backward_mode": th.BackwardMode.TRUNCATED,
+            "backward_mode": "TRUNCATED",
```
Comment (mhmukadam) [inline theseus/optimizer/nonlinear/tests/test_backwards.py:114]: nit: TRUNCATED -> truncated

**Review by mhmukadam — APPROVED** [review]
Looks great!

File: theseus/optimizer/nonlinear/tests/test_backwards.py (hunk 2 above)
Comment (luisenp) [reply theseus/optimizer/nonlinear/tests/test_backwards.py:114]: Oh, I was just adding a call with uppercase to add coverage for this case.

**Pull request merged**